common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -652,6 +829,274 @@ class NativeAuthMsalController : BaseNativeAuthController() {
+    }
+
+    @VisibleForTesting
+    fun performSignUpStartUsingPasswordRequest(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpStartUsingPasswordCommandParameters
+    ): SignUpStartApiResult {
+        return oAuth2Strategy.performSignUpStartUsingPassword(
+            commandParameters = parameters
+        )
+    }
+
+    private fun performSignUpChallengeCall(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        signupToken: String
+    ): SignUpChallengeApiResult {
+        return oAuth2Strategy.performSignUpChallenge(signUpToken = signupToken)
+    }
+
+    private fun performSignUpSubmitCode(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpSubmitCodeCommandParameters
+    ): SignUpContinueApiResult {
+        return oAuth2Strategy.performSignUpSubmitCode(commandParameters = parameters)
+    }
+    private fun performSignUpSubmitPassword(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpSubmitPasswordCommandParameters
+    ): SignUpContinueApiResult {
+        return oAuth2Strategy.performSignUpSubmitPassword(commandParameters = parameters)
+    }
+
+    @VisibleForTesting
+    fun performSignUpSubmitUserAttributes(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpSubmitUserAttributesCommandParameters
+    ): SignUpContinueApiResult {
+        return oAuth2Strategy.performSignUpSubmitUserAttributes(commandParameters = parameters)
+    }
+
+    private fun SignUpChallengeApiResult.toSignUpStartCommandResult(): SignUpStartCommandResult {
+        return when (this) {
+            is SignUpChallengeApiResult.OOBRequired -> {
+                SignUpCommandResult.CodeRequired(
+                    signupToken = this.signupToken,
+                    codeLength = this.codeLength,
+                    challengeTargetLabel = this.challengeTargetLabel,
+                    challengeChannel = this.challengeChannel
+                )
+            }
+            is SignUpChallengeApiResult.PasswordRequired -> {
+                SignUpCommandResult.PasswordRequired(
+                    signupToken = this.signupToken
+                )
+            }
+            SignUpChallengeApiResult.Redirect -> {
+                INativeAuthCommandResult.Redirect()
+            }
+            is SignUpChallengeApiResult.ExpiredToken, is SignUpChallengeApiResult.UnsupportedChallengeType,
+            is SignUpChallengeApiResult.UnknownError -> {
+                Logger.warn(
+                    TAG,
+                    "Unexpected result: $this"
+                )
+                this as ApiErrorResult
+                INativeAuthCommandResult.UnknownError(
+                    error = this.error,
+                    errorDescription = this.errorDescription,
+                    details = this.details
+                )
+            }
+        }
+    }
+
+    private fun SignUpContinueApiResult.toSignUpSubmitCodeCommandResult(
```

Comment:
Please add some comments describing why this method is needed

Reply:
Done